### ISSUE TITLE
fix support for obj parameter of Class.from_parent

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [pytest-dev/pytest#7063]
in https://github.com/pytest-dev/pytest/commit/c99c7d0f95170a5e308ed6c37e63b2d90794a4f9#r38397315 @blueyed noted a ignored optional parameter

as far as i remember this is a refactoring artifact i accidentally left,
i should ensure it can be used

**Comment by blueyed** [comment]
`Class` never took an `obj`.
I've noticed this when optimizing things to use passed-in objects.

**Comment by RonnyPfannschmidt** [comment]
@blueyed as i said its a refactoring artifact i accidentally left

 it will be revisited as pretty much everything should try to use passed in objects

**Comment by blueyed** [comment]
> it will be revisited as pretty much everything should try to use passed in objects

Yeah.  I've came across this when noticing that typically both `FunctionDefinition` and `Function` are being created - both creating the obj etc.

As for your "please coordinate" comment on https://github.com/pytest-dev/pytest/pull/7052 you might want to provide some reference to why `FunctionDefinition` is needed / what's the plan with it.

**Comment by RonnyPfannschmidt** [comment]
Functiondefinition is the node which represents a function without parameterization, its collect method should run generate_tests

**Comment by blueyed** [comment]
>  i should ensure it can be used

Will it be named `obj` then still?  `Function` uses `callobj` for it currently/already.

**Comment by RonnyPfannschmidt** [comment]
yes, , obj is the intenral/semantical name used in practically all nodes, the function object using callobj will deprecate/disappear with the introduction of functiondefinition in the collection tree
